### PR TITLE
4.7.4 - v4 Migration Cleanup

### PR DESCRIPTION
**Pull request by rejahrehim**
This PR part of issue #83
- [ ] This PR handles the issue and requires no additional PRs.
- [x] You have validated the need for this change.

**What did this PR accomplish?**

- Fix linting

### CODE REVIEW

File: document/4_Web_Application_Security_Testing/4.7_Session_Management_Testing/4.7.4_Testing_for_Exposed_Session_Variables_OTG-SESS-004.md [hunk 1]
```diff
@@ -1,84 +1,89 @@
...
 
-Note that if there is also an element to the site where the user is tracked with Session IDs but security is not present (e.g., noting which public documents a registered user downloads) it is essential that a different Session ID is used. The Session ID should therefore be monitored as the client switches from the secure to non-secure elements to ensure a different one is used.\
+Note that if there is also an element to the site where the user is tracked with Session IDs but security is not present (e.g., noting which public documents a registered user downloads) it is essential that a different Session ID is used. The Session ID should therefore be monitored as the client switches from the secure to non-secure elements to ensure a different one is used.
 
-**Result Expected:**\
+**Result Expected:**
```
[the remark this answers is not in the document]
Comment (ThunderSon) [reply document/4_Web_Application_Security_Testing/4.7_Session_Management_Testing/4.7.4_Testing_for_Exposed_Session_Variables_OTG-SESS-004.md]: Consider this for the other ones as well

File: document/4_Web_Application_Security_Testing/4.7_Session_Management_Testing/4.7.4_Testing_for_Exposed_Session_Variables_OTG-SESS-004.md [hunk 2]
```diff
@@ -1,84 +1,89 @@
...
 Every time the authentication is successful, the user should expect to receive:
 
--   A different session token
--   A token sent via encrypted channel every time they make an HTTP Request
+- A different session token
+- A token sent via encrypted channel every time they make an HTTP Request
+
+### Testing for Proxies & Caching Vulnerabilities:
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.7_Session_Management_Testing/4.7.4_Testing_for_Exposed_Session_Variables_OTG-SESS-004.md]: Remove trailing `:` for all sections

File: document/4_Web_Application_Security_Testing/4.7_Session_Management_Testing/4.7.4_Testing_for_Exposed_Session_Variables_OTG-SESS-004.md [hunk 3]
```diff
@@ -1,84 +1,89 @@
...
+- HTTP Headers
+- Message Body (e.g., POST or page content)
 
 Each time Session ID data is passed between the client and the server, the protocol, cache, and privacy directives and body should be examined. Transport security here refers to Session IDs passed in GET or POST requests, message bodies, or other means over valid HTTP requests.
 
-How to Test
------------
+## How to Test
+
+### Testing for Encryption & Reuse of Session Tokens Vulnerabilities:
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.7_Session_Management_Testing/4.7.4_Testing_for_Exposed_Session_Variables_OTG-SESS-004.md]: Trailing `:`

**Comment by ThunderSon** [comment]
One last comment is still not resolved

File: document/4_Web_Application_Security_Testing/4.7_Session_Management_Testing/4.7.4_Testing_for_Exposed_Session_Variables_OTG-SESS-004.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (rejahrehim) [reply document/4_Web_Application_Security_Testing/4.7_Session_Management_Testing/4.7.4_Testing_for_Exposed_Session_Variables_OTG-SESS-004.md]: Could you please help me with this comment. I didn't get that.

[the remark this answers is not in the document]
Comment (rejahrehim) [reply document/4_Web_Application_Security_Testing/4.7_Session_Management_Testing/4.7.4_Testing_for_Exposed_Session_Variables_OTG-SESS-004.md]: @ThunderSon Updated with fix

File: document/4_Web_Application_Security_Testing/4.7_Session_Management_Testing/4.7.4_Testing_for_Exposed_Session_Variables_OTG-SESS-004.md [hunk 4]
```diff
@@ -1,84 +1,86 @@
...
+In general, the Session ID should never be sent over unencrypted transport and should never be cached. The application should be examined to ensure that encrypted communications are both the default and enforced for any transfer of Session IDs. Furthermore, whenever the Session ID is passed, directives should be in place to prevent its caching by intermediate and even local caches.
 
 The application should also be configured to secure data in caches over both HTTP/1.0 and HTTP/1.1 – RFC 2616 discusses the appropriate controls with reference to HTTP. HTTP/1.1 provides a number of cache control mechanisms. Cache-Control: no-cache indicates that a proxy must not re-use any data. Whilst Cache-Control: Private appears to be a suitable directive, this still allows a non-shared proxy to cache data. In the case of web-cafes or other shared systems, this presents a clear risk. Even with single-user workstations the cached Session ID may be exposed through a compromise of the file-system or where network stores are used. HTTP/1.0 caches do not recognise the Cache-Control: no-cache directive.
```
[the remark this answers is not in the document]
Comment (rejahrehim) [reply document/4_Web_Application_Security_Testing/4.7_Session_Management_Testing/4.7.4_Testing_for_Exposed_Session_Variables_OTG-SESS-004.md]: @ThunderSon  Done

File: document/4_Web_Application_Security_Testing/4.7_Session_Management_Testing/4.7.4_Testing_for_Exposed_Session_Variables_OTG-SESS-004.md [hunk 5]
```diff
@@ -1,84 +1,86 @@
...
+If the Session ID could be presented by an attacker to the application to gain access, then it must be protected in transit to mitigate that risk. It should therefore be ensured that encryption is both the default and enforced for any request or response where the Session ID is passed, regardless of the mechanism used (e.g., a hidden form field). Simple checks such as replacing <https://> with <http://> during interaction with the application should be performed, together with modification of form posts to determine if adequate segregation between the secure and non-secure sites is implemented.
```
[the remark this answers is not in the document]
Comment (rejahrehim) [reply document/4_Web_Application_Security_Testing/4.7_Session_Management_Testing/4.7.4_Testing_for_Exposed_Session_Variables_OTG-SESS-004.md]: @ThunderSon  Done

File: document/4_Web_Application_Security_Testing/4.7_Session_Management_Testing/4.7.4_Testing_for_Exposed_Session_Variables_OTG-SESS-004.md [hunk 6]
```diff
@@ -35,7 +35,7 @@ Proxies must also be considered when reviewing application security. In many cas
...
+The application should also be configured to secure data in caches over both HTTP/1.0 and HTTP/1.1 – RFC 2616 discusses the appropriate controls with reference to HTTP. HTTP/1.1 provides a number of cache control mechanisms. `Cache-Control: no-cache` indicates that a proxy must not re-use any data. Whilst `Cache-Control: Private` appears to be a suitable directive, this still allows a non-shared proxy to cache data. In the case of web-cafes or other shared systems, this presents a clear risk. Even with single-user workstations the cached Session ID may be exposed through a compromise of the file-system or where network stores are used. HTTP/1.0 caches do not recognise the `Cache-Control: no-cache` directive.
 
 > The “Expires: 0” and Cache-Control: max-age=0 directives should be used to further ensure caches do not expose the data. Each request/response passing Session ID data should be examined to ensure appropriate cache directives are in use.
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.7_Session_Management_Testing/4.7.4_Testing_for_Exposed_Session_Variables_OTG-SESS-004.md]: Please set the key-value pairs in backticks like you did in the previous section

Comment (rejahrehim) [reply document/4_Web_Application_Security_Testing/4.7_Session_Management_Testing/4.7.4_Testing_for_Exposed_Session_Variables_OTG-SESS-004.md]: @ThunderSon Fixed.

**Pull request merged**